transmart/api/v2/data_structures.py

```diff
@@ -119,6 +122,27 @@ def pretty(self, root=None, depth=0, spacing=2):
         for child in root.get('children', {}):
             s += "\n%s" % self.pretty(child, depth + 1, spacing)
         return s
+    
+    def create_dataframe(self, root=None):
+        """
+        Create dataframe representation of tree.
+        """
+        
+        if root is None:
+            top_nodes = []
+            for root in self.json.get('tree_nodes'):
+                top_nodes += self.create_dataframe(root)
+            return top_nodes
+
+        thisnode = root.copy()
+        thisnode.pop("children", None)
```

Comment:
my guess is you would not need to copy if you pop the children after recursing them.